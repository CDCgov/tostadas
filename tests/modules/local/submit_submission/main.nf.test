nextflow_process {

    name "Test Process SUBMISSION"
    script "modules/local/submit_submission/main.nf"
    process "SUBMISSION"

    test("Run with profile test, virus | biosample") {

        when {
            params {
                // define gold standard output(s) for comparison
                gs_biosample_xml = "${projectDir}/modules/local/prep_submission/tests/submission_biosample.xml"
                // define parameters here
                submission_name = "batch_1"
                identifier = "mpxv_test_metadata"
                submission_mode = "ftp"
                test = true
                send_submission_email = false
                dryrun = true
            }

            process {
                """
                // define inputs of the process here
                input[0] = path("${params.submission_name}")
                input[1] = path("${projectDir}/conf/submission_config.yaml")
                """
            }
        }

        then {
            assert process.success
            // todo: add more tests

        }
    }
}
