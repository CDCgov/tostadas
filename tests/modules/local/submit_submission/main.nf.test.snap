{
    "Run with profile test, virus": {
        "content": [
            "submission.log:md5,24d4960c2f0ef871aa264aa779dd0373"
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.6"
        },
        "timestamp": "2025-08-28T16:22:39.009561376"
    }
}